service/resource/deployment/spec.go

```diff
@@ -2,11 +2,16 @@ package deployment
 
 // Deployment defines an Azure Deployment that deploys an ARM template.
 type Deployment struct {
-	Name            string
-	Parameters      map[string]interface{}
-	ResourceGroup   string
-	TemplateURI     string
-	TemplateVersion string
+	Name          string
+	Parameters    map[string]interface{}
+	ResourceGroup string
+	TemplateURI   string
+
+	// TmplateContentVersion is value to fill in
+	// github.com/Azure/azure-sdk-for-go/arm/resources/resources.TemplateLink.ContentVersion.
+	// For more information see contentVersion documentation
+	// https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-authoring-templates.
+	TemplateContentVersion string
```

Comment:
I did my best to describe it. I don't know why we have to specify this in both ARM template and sdk Deployment struct. This is confusing and it's the reason we have that here.